source/lac/trilinos_sparse_matrix.cc

```diff
@@ -2651,6 +2726,114 @@ namespace TrilinosWrappers
+        , domain_map(
+            payload_exemplar.locally_owned_domain_indices().make_trilinos_map(
+              communicator.Comm()))
+        , range_map(
+            payload_exemplar.locally_owned_range_indices().make_trilinos_map(
+              communicator.Comm()))
+      {
+        vmult = [&payload_exemplar, &preconditioner](Range &       tril_dst,
+                                                     const Domain &tril_src) {
+          // Duplicated from TrilinosWrappers::PreconditionBase::vmult
+          // as well as from TrilinosWrappers::SparseMatrix::Tvmult
+          Assert(&tril_src != &tril_dst,
+                 TrilinosWrappers::SparseMatrix::ExcSourceEqualsDestination());
+          internal::check_vector_map_equality(payload_exemplar,
+                                              tril_src,
+                                              tril_dst,
+                                              payload_exemplar.UseTranspose());
+          internal::check_vector_map_equality(
+            preconditioner.trilinos_operator(),
+            tril_src,
+            tril_dst,
+            preconditioner.trilinos_operator().UseTranspose());
+
+          const int ierr =
+            preconditioner.trilinos_operator().ApplyInverse(tril_src, tril_dst);
+          AssertThrow(ierr == 0, ExcTrilinosError(ierr));
+        };
+
+        Tvmult = [&payload_exemplar, &preconditioner](Domain &     tril_dst,
+                                                      const Range &tril_src) {
+          // Duplicated from TrilinosWrappers::PreconditionBase::vmult
+          // as well as from TrilinosWrappers::SparseMatrix::Tvmult
+          Assert(&tril_src != &tril_dst,
+                 TrilinosWrappers::SparseMatrix::ExcSourceEqualsDestination());
+          internal::check_vector_map_equality(payload_exemplar,
+                                              tril_src,
+                                              tril_dst,
+                                              !payload_exemplar.UseTranspose());
+          internal::check_vector_map_equality(
+            preconditioner.trilinos_operator(),
+            tril_src,
+            tril_dst,
+            !preconditioner.trilinos_operator().UseTranspose());
+
+          preconditioner.trilinos_operator().SetUseTranspose(
+            !preconditioner.trilinos_operator().UseTranspose());
+          const int ierr =
+            preconditioner.trilinos_operator().ApplyInverse(tril_src, tril_dst);
+          AssertThrow(ierr == 0, ExcTrilinosError(ierr));
+          preconditioner.trilinos_operator().SetUseTranspose(
+            !preconditioner.trilinos_operator().UseTranspose());
+        };
+
+        inv_vmult = [&payload_exemplar,
+                     &preconditioner](Domain &tril_dst, const Range &tril_src) {
+          // Duplicated from TrilinosWrappers::PreconditionBase::vmult
+          // as well as from TrilinosWrappers::SparseMatrix::Tvmult
```

Comment:
This functions appears to be a copy of the `vmult` function 40 lines above. Do we have a possibility to redirect `inv_vmult` to the same implementation in order to avoid code duplication? Same for `inv_Tvmult`.